docs/community/tutorials-contest/2022/precompile-storage-gateway/README.md

```diff
@@ -0,0 +1,1222 @@
+
+	/* Call precompiled contract. */
+	// if iszero(staticcall(not(0), 0x53B, add(pkg, 32), 0xd5, stored, 0x40)) {
+	if iszero(call(gas(), 0x53b, 0, 0, 0, pkg, 20)) {
+	  invalid()
+	}
+
+	/* Load stored data. */
+	stored := mload(freemem)
+  }
+
+  /* Return stored data. */
+  return stored;
+}
+```
+
+#### CID
+
+__Content Identifier__
+
+Type: `uint256`
+
+This will be used to query the respective storage network for the user's requested data.
+
+#### Network ID
+
+Type: `uint8`
+
+Specifies which storage network to query.
+
+#### IsClustered
+
+Type: `bool`
+
+Will allow the request to be made to multiple nodes and validate the results before returning the data.
+
+### `saveData` function
+
+`Type: Function`
+
+This will reside at `0x53B`, and provide a bridge to the requested storage network, taking as input, in order:
+- __CID__ [`uint256`] _(content identifier)_
+- __Network ID__ [`uint8`] _(ie. IPFS, Storj, Sia, etc)_
+- __IsClustered__ [`bool`] _(default is 2-of-3 nodes)_
+- __Data__ [`string`] _(content to store)_
+
+```js
+/**
+ * Set Storage
+ *
+ * User specifies a data identifier and network to query. Validators will
+ * connect to the respective network and retrieve the data.
+ */
+function setStorage(
+  uint256 _cid,
+  bytes8 memory _network,
+  bool _isClustered,
+  string calldata _data
+) public view returns (bytes memory) {
+  /* Initialize stored data holder. */
+  bytes memory stored;
+
+  /* Build request package. */
+  bytes memory pkg = abi.encodePacked(
+	_cid,
+	_network,
+	_isClustered,
+	_data
+  );
+
+  /* Perform assembly action. */
+  assembly {
+	/* Initialize free memory. */
+	let freemem := mload(0x40)
+
+	/* Append package to memory. */
+	let pkg := add(freemem, 12)
+
+	/* Call precompiled contract. */
+	// if iszero(staticcall(not(0), 0x53B, add(pkg, 32), 0xd5, stored, 0x40)) {
+	if iszero(call(gas(), 0x53b, 0, 0, 0, pkg, 20)) {
+	  invalid()
+	}
+
+	/* Load stored data. */
+	stored := mload(freemem)
+  }
+
+  /* Return stored data. */
+  return stored;
+}
+```
+
+#### CID
+
+__Content Identifier__
+
+Type: `uint256`
+
+This will be used to query the respective storage network for the user's requested data.
+
+#### Network ID
+
+Type: `uint8`
+
+Specifies which storage network to query.
+
+#### IsClustered
+
+Type: `bool`
+
+Will allow the request to be made to multiple nodes and validate the results before returning the data.
+
+#### Data
+
+Type: `string`
+
+This is the content that you wish to write to the external storage.
+
+---
+
+## Cautions
+
+It's important to be patient when you first start implementing Precompile contracts. This is especially true if you're a first-time Golang user and may not be familiar with the language syntax.
+
+The following are a few of the most common issues faced by the Subnet developer community.
+
+### Troubleshooting
+
+There are still issues that need to be solved, related to:
+
+1. Contract permissions
+2. Request timeout
+
+#### Contract permissions
```

Comment:
All section header should be title case, please check and update all section headers with https://titlecase.com/

Reply:
done!